cli/src/main/java/com/devonfw/tools/ide/merge/xmlmerger/XmlMergeSupport.java

```diff
@@ -0,0 +1,332 @@
+import java.util.function.Predicate;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.QName;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Attr;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Utility class with static helper methods for XML merging.
+ */
+public interface XmlMergeSupport {
+
+  /**
+   * The XML namespace used by this {@link XmlMerger} to configure merge specific meta-information.
+   *
+   * @see XmlMergeSupport#hasMergeNamespace(Attr)
+   */
+  String MERGE_NS_URI = "https://github.com/devonfw/IDEasy/merge";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_ID = "id";
+
+  /** XPath for {@link Attr} {@value} */
+  String XPATH_ATTR_ID = "@id";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_NAME = "name";
+
+  /** XPath for {@link Attr} {@value} */
+  String XPATH_ATTR_NAME = "@name";
+
+  /** XPath for {@link Element#getTagName() element tag name}: {@value} */
+  String XPATH_ELEMENT_NAME = "name()";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_STRATEGY = "strategy";
+
+  /** XPath for {@link Element#getTextContent() element text}: {@value} */
+  String XPATH_ELEMENT_TEXT = "text()";
+
+  /**
+   * @param attribute the XML {@link Attr} to check.
+   * @return {@code true} if the {@code attribute} is specific for our XML merger and related to {@link XmlMergeSupport#MERGE_NS_URI} (that needs to be omitted
+   *     in the XML output), {@code false} otherwise (regular {@code attribute} for the target payload).
+   */
+  static boolean hasMergeNamespace(Attr attribute) {
+
+    if (MERGE_NS_URI.equals(attribute.getNamespaceURI())) {
+      return true;
+    } else {
+      return "xmlns".equals(attribute.getPrefix()) && MERGE_NS_URI.equals(attribute.getValue());
+    }
+  }
+
+  /**
+   * @param element the {@link Element} for which the XPath is requested.
+   * @return the XPath expression.
+   */
+  static String getXPath(Element element) {
+
+    return getXPath(element, false);
+  }
+
+  /**
+   * @param element the {@link Element} for which the XPath is requested.
+   * @param includeAttributes {@code true} to also include the attributes of
+   * @return the XPath expression.
+   */
+  static String getXPath(Element element, boolean includeAttributes) {
+
+    StringBuilder sb = new StringBuilder();
+    getXPath(sb, element, includeAttributes);
+    return sb.toString();
+  }
+
+  /**
+   * @param sb the {@link StringBuilder} where to {@link StringBuilder#append(String) append} the XML.
+   * @param element the {@link Element} with the XPath to append.
+   */
+  private static void getXPath(StringBuilder sb, Element element, boolean includeAttributes) {
```

Comment:
Please adjust Javadoc, includeAttributes is missing.

Reply:
Its a private method and the public `getXPath` method with such arguments is just above.
I just removed the JavaDoc as every developer should be able to read and understand the public method and then also get the private one called from the public method.
Having all this Copy&Paste of JavaDoc is IMHO not always making things better esp. when something changes and only parts of the redundant JavaDoc gets updated.